applications/Chat/evaluate/metric.py

```diff
@@ -0,0 +1,177 @@
+    weights = [
+        (1./1., 0., 0., 0.),
+        (1./2., 1./2., 0., 0.),
+        (1./3., 1./3., 1./3., 0.),
+        (1./4., 1./4., 1./4., 1./4.)
+     ]
+
+    for pred, target in zip(preds, targets):
+        pred_list = (' '.join(jieba.cut(pred))).split()
+        target_list = [(' '.join(jieba.cut(target))).split()]
+
+        bleu = sentence_bleu(target_list, pred_list, weights = weights)
+        cumulative_bleu = [a+b for a, b in zip(cumulative_bleu, bleu)]
+
+    bleu_scores["bleu1"] = cumulative_bleu[0]/len(preds)
+    bleu_scores["bleu2"] = cumulative_bleu[1]/len(preds)
+    bleu_scores["bleu3"] = cumulative_bleu[2]/len(preds)
+    bleu_scores["bleu4"] = cumulative_bleu[3]/len(preds)
+
+    return bleu_scores
+
+
+def rouge_cn_score(preds: list, targets: list) -> dict:
+    """Calculate Chinese ROUGE Score Metric
+
+    The calculation includes ROUGE-1 for unigram, ROUGE-2 for bigram
+    and ROUGE-L. ROUGE-N evaluates the number of matching n-grams between 
+    the preds and targets. ROUGE-L measures the number of matching
+    longest common subsequence (LCS) between preds and targets.
+    """
+    rouge_scores = {"rouge1": {}, "rouge2": {}, "rougeL": {}}
+    all_preds = []
+    all_targets = []
+
+    for pred, target in zip(preds, targets):
+        pred_list = ' '.join(jieba.cut(pred))
+        target_list = ' '.join(jieba.cut(target))
+        all_preds.append(pred_list)
+        all_targets.append(target_list)
+
+    rouge_cn = Rouge_cn()
+    rouge_avg = rouge_cn.get_scores(all_preds, all_targets, avg=True)
+
+    rouge_scores["rouge1"] = rouge_avg["rouge-1"]
+    rouge_scores["rouge2"] = rouge_avg["rouge-2"]
+    rouge_scores["rougeL"] = rouge_avg["rouge-l"]
+
+    return rouge_scores
+
+
+def distinct_score(preds: list) -> dict:
+    """Calculate Distinct Score Metric
+
+    This metric refers to https://arxiv.org/abs/1510.03055.
+    It evaluates the diversity of generation text by counting
+    the unique n-grams.
+    """
+    distinct_score = {"distinct": 0}
+    cumulative_distinct = []
+
+    for pred in preds:
+        pred_seg_list = list(' '.join(jieba.cut(pred)))
+        count_segs = len(pred_seg_list)
+        unique_segs = set(pred_seg_list)
+        count_unique_chars = len(unique_segs)
+
+        cumulative_distinct.append(count_unique_chars/count_segs)
+
+    distinct_score["distinct"] = statistics.mean(cumulative_distinct)
+
+    return distinct_score
+
+
+def bert_score(preds: list, targets: list) -> dict:
+    """Calculate BERTScore Metric
+
+    The BERTScore evaluates the semantic similarity between 
+    tokens of preds and targets with BERT.
+    """
+    bert_score = {"precision": 0, "recall": 0, "f1_score": 0}
+    pred_list = []
+    target_list = []
+
+    for pred, target in zip(preds, targets):
+        pred_list.append(' '.join(jieba.cut(pred)))
+        target_list.append(' '.join(jieba.cut(target)))
+    
+    P, R, F = score(pred_list, target_list, lang="zh", verbose=True)
```

Comment:
Do we need to add another parameter `language` in this fuction and use an `if` to decide which parameter `lang` will take.

For example:
```python
if language == "cn":
    P, R, F = score(pred_list, target_list, lang="zh", verbose=True)
elif language == "en":
    P, R, F = score(pred_list, target_list, lang="en", verbose=True)
```